spe.py

```diff
@@ -0,0 +1,318 @@
+#!/usr/bin/env python
+'''
+Do the Single Photoelectron anaylsis
+
+Usage:
+  spe.py [options] <files>...
+
+Options:
+  -h --help     Show this screen.
+'''
+
+from docopt import docopt
+from digicampipe.io.event_stream import event_stream
+import numpy as np
+import matplotlib.pyplot as plt
+from histogram.histogram import Histogram1D
+from tqdm import tqdm
+import peakutils
+from  scipy.signal import find_peaks_cwt
+from digicampipe.utils.utils import integrate
+from scipy import ndimage
+import scipy
+from iminuit import Minuit
+from digicampipe.io.containers import CalibrationContainer
+from digicampipe.utils.utils import filter_template
+
+
+def calibration_event_stream(path, telescope_id, max_events):
+
+    container = CalibrationContainer()
+
+    for event in event_stream(path, max_events=max_events):
+
+        container.adc_samples = event.r0.tel[telescope_id].adc_samples
+        container.n_pixels = container.adc_samples.shape[0]
+
+        yield container
+
+
+def compute_event_stats(events):
+
+    for count, event in enumerate(events):
+
+        if count == 0:
+
+            n_pixels = event.n_pixels
+            adc_histo = Histogram1D(data_shape=(n_pixels, ), bin_edges=np.arange(0, 4095, 1), axis_name='[LSB]')
+
+        adc_histo.fill(event.adc_samples)
+
+    mode = adc_histo.mode()
+    mean = adc_histo.mean()
+    std = adc_histo.std()
+    max = adc_histo.max
+
+    return mean, std, mode, max
+
+
+def subtract_baseline(events, value):
+
+    for count, event in enumerate(events):
+
+        event.adc_samples = event.adc_samples - value[:, np.newaxis]
```

Comment:
I'd always write this:
```
event.adc_samples = event.adc_samples - value[:, np.newaxis]
```
as:
```
event.adc_samples -= value[:, np.newaxis]
```

Because in the former variant, the reader has to pay close attention if the variable names on both sides of the `=`-sign are the same or not.

Reply:
ok I tried to do this at some point. This is what I got:

```bash
    event.adc_samples -= baseline[..., np.newaxis]
TypeError: Cannot cast ufunc subtract output from dtype('float32') to dtype('uint16') with casting rule 'same_kind'
```

so in the end I need to convert `event.adc_samples` to float32 to make it work.

Reply:
Ah so this means this: 
```
event.adc_samples = event.adc_samples - value[:, np.newaxis]
```
does not only subtract a values from `event.adc_samples` but it also **secretly** changed the dtype of adc_samples. That is quite tricky for a reader to guess at this point.